q2_fmt/_peds.py

```diff
@@ -0,0 +1,147 @@
+    # TODO: Make incomplete samples possible move this to heatmap
+    metadata = metadata.to_dataframe()
+
+    try:
+        num_timepoints = metadata[time_column].dropna().unique().size
+    except Exception:
+        raise KeyError('There was and error finding %s in the metadata'
+                       % time_column)
+    try:
+        metadata[time_column].dropna().astype(int)
+    except Exception:
+        raise KeyError('%s must be numeric' % time_column)
+    try:
+        subject_occcurance_df = (metadata[subject_column].value_counts()
+                                 .to_frame())
+    except Exception:
+        raise KeyError('There was and error finding %s in the metadata'
+                       % subject_column)
+    if (subject_occcurance_df[subject_column] != num_timepoints).any():
+        if drop_incomplete_subjects:
+            subject_to_keep = (subject_occcurance_df
+                               .loc[subject_occcurance_df[subject_column]
+                                    == num_timepoints].index)
+            metadata = metadata[metadata[subject_column].isin(subject_to_keep)]
+        else:
+            incomplete_subjects = (subject_occcurance_df
+                                   .loc[subject_occcurance_df[subject_column]
+                                        < num_timepoints].index).to_list()
+            raise KeyError('Missing timepoints for associated subjects.'
+                           ' Please make sure that all subjects have all'
+                           ' timepoints or use drop_incomplete_subjects'
+                           ' parameter. The incomplete subjects were %s'
+                           % incomplete_subjects)
+
+    try:
+        reference_series = metadata[reference_column]
+    except Exception:
+        raise KeyError('There was and error finding %s in the metadata'
+                       % reference_column)
+
+    used_references = reference_series[~metadata[time_column].isna()]
+
+    if used_references.isna().any():
+        if filter_missing_references:
+            used_references = used_references.dropna()
+        else:
+            nan_references = used_references.index[used_references.isna()]
+            raise KeyError('Missing references for the associated sample data.'
+                           ' Please make sure that all samples with a'
+                           ' timepoint value have an associated reference.'
+                           ' IDs where missing references were found:'
+                           ' %s' % (tuple(nan_references),))
+
+    peds_df = _compute_peds(used_references, table, metadata, time_column,
+                            reference_column, subject_column)
+    return peds_df
+
+
+def _compute_peds(reference_series: pd.Series, table: pd.Series,
+                  metadata: qiime2.Metadata, time_column: str,
+                  reference_column: str,
+                  subject_column: str) -> (pd.DataFrame):
+
+    PEDSserieslist = []
```

Comment:
This is probably a matter of style/preference over anything else - but I'm seeing some inconsistencies in your variable assignment names from snake case to camel case (`variable_example_a` vs. `variableExampleA`). I think it might look nicer to have everything snake case (since that's the typical casing for variable assignment across the framework) - but like I said before, it's more of a style/preference thing than an actual issue with your code functionality or readability. @ebolyen do you think it matters one way or the other?